docs/source/model_doc/character_bert.rst

```diff
@@ -0,0 +1,114 @@
+    an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+    specific language governing permissions and limitations under the License.
+
+CharacterBERT
+-----------------------------------------------------------------------------------------------------------------------
+
+Overview
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The CharacterBERT model was proposed in `CharacterBERT: Reconciling ELMo and BERT for Word-Level Open-Vocabulary Representations From Characters <https://www.aclweb.org/anthology/2020.coling-main.609>`__  by Hicham EL BOUKKOURI, Olivier FERRET, Thomas LAVERGNE, Hiroshi NOJI, Pierre ZWEIGENBAUM and Junichi TSUJII. In short, it is a variant of BERT that uses a Character-CNN module instead of WordPieces to achieve a word-level, open-vocabulary model that is more convenient to use as well as more robust to noisy (misspellings) inputs.
+
+The abstract from the paper is the following:
+
+*Due to the compelling improvements brought by BERT, many recent representation models adopted the Transformer architecture as their main building block, consequently inheriting the wordpiece tokenization system despite it not being intrinsically linked to the notion of Transformers. While this system is thought to achieve a good balance between the flexibility of characters and the efficiency of full words, using predefined wordpiece vocabularies from the general domain is not always suitable, especially when building models for specialized domains (e.g., the medical domain). Moreover, adopting a wordpiece tokenization shifts the focus from the word level to the subword level, making the models conceptually more complex and arguably less convenient in practice. For these reasons, we propose CharacterBERT, a new variant of BERT that drops the wordpiece system altogether and uses a Character-CNN module instead to represent entire words by consulting their characters. We show that this new model improves the performance of BERT on a variety of medical domain tasks while at the same time producing robust, word-level, and open-vocabulary representations.*
+
+Tips:
+
+- CharacterBert is a token-level model that considers each token as a sequence of characters. Therefore, the input shape is (batch dimension, tokens dimension, characters dimension) instead of (batch dimension, tokens dimension).
+- Because the model looks at the characters of each input token, it can be more robust to noise/misspellings than BERT. 
+- Because CharacterBert does not rely on a vocabulary of wordpieces, the same model can be seamlessly re-used in different domains or languages (But keep in mind that it was pre-trained on English texts).
+- If you are working in the medical domain, make sure you use the version of CharacterBERT that is pre-trained on medical corpora.
+
```

Comment:
Add a line here with the contributor that added the model as well as where the original code is (see another doc page for an example).